src/sync/partial_sync.cpp

```diff
@@ -33,6 +36,65 @@
 #include <realm/lang_bind_helper.hpp>
 #include <realm/util/scope_exit.hpp>
 
+using namespace std::chrono;
+
+namespace {
+
+    // FIXME: Replace these methods with: https://github.com/realm/realm-core/pull/3259
+
+    /// Turn a system time point value into the 64-bit integer representing ns since the Unix epoch.
+    int64_t ns_since_unix_epoch(const system_clock::time_point& point)
+    {
+        return duration_cast<nanoseconds>(point.time_since_epoch()).count();
+    }
+
+    realm::Timestamp timestamp_now() {
+        int64_t ns_since_epoch = ns_since_unix_epoch(system_clock::now());
+        int64_t s_arg = ns_since_epoch / static_cast<int64_t>(realm::Timestamp::nanoseconds_per_second);
+        int32_t ns_arg = ns_since_epoch % realm::Timestamp::nanoseconds_per_second;
+        return realm::Timestamp(s_arg, ns_arg);
+    }
+
+    // Calculates the expiry date, claming at the high end if a timestamp overflows
+    realm::Timestamp calculate_expiry_date(realm::Timestamp starting_time, int64_t user_ttl_ms) {
```

Comment:
Newline before `{` for function definitions (and nothing else). A bunch of these in the file.